src/puppetlabs/puppetdb/command.clj

```diff
@@ -1032,4 +1050,16 @@
    [this]
    (throw-unless-ready (service-context this))
    (throw-if-shutdown-pending (get-shutdown-reason))
-   (-> this service-context :response-mult)))
+   (-> this service-context :response-mult))
+
+  (pause-execution
+    [this]
+    (let [context (service-context this)
+          blocked? (context :blocked?)]
+      (pause-command-service blocked?)))
+
+  (resume-execution
+    [this]
+    (let [context (service-context this)
+          blocked? (context :blocked?)]
+      (resume-command-service blocked?))))
```

Comment:
Not critical, but could rewrite these a bit -- one option: `(pause-command-service (-> this service-context :blocked?))`.